### ISSUE TITLE
Automate TLS certificate renewal

### ISSUE THREAD

**Issue by bickelj** [PhilanthropyDataCommons/deploy#64]
Alternatively, renew by Mothers' Day 2023
<details><summary>Old</summary><details>Alternatively, automate via cron or some other means. The reason it is not automated yet is the interaction between

    nginx in a container
    certbot needing port 80
    port 80 needs root
</details>

**Comment by bickelj** [comment]
The last concern may not be an obstacle. I just changed the internal port on the reverse proxy to 443 from 8443 and had no trouble, despite the reverse-proxy user not being root. A `docker-proxy` process running as root handles the translation and the reverse-proxy container is OK to serve on 443. It can probably serve on 80 as well.

The next question, then, is do we want to run certbot in the same container as nginx? Probably not. Then what? Off to the side? It doesn't need to run all the time, just intermittently. Totally off to the side via cron? If running on the host, sure, but if running in a container, probably not using cron. Then what? If running certbot outside the container whether in another container or on the host, we need to signal nginx to reload. [Ofelia](https://github.com/mcuadros/ofelia) looks promising if we go fully containerized. The fastest solution is probably to run the currently-used commands via the crontab. That is the next step.

Immediately the question arises about how to get feedback from the renewal process, then. I suppose a chat message would be good, except now we need to make sure the script that cron runs would have access to those environment variables. We already have the chat bot configured in environment variables accessible and passed on to containers by the compose script. So if we ran as part of that same compose script, sending a message to chat would not require a separate configuration. Maybe fully containerized is the right way.

Consider a local dev machine, though. We want to be able to generate a self-signed cert locally, point the compose script to it via `.env`, and try things out. Also the (certbot docs)[https://eff-certbot.readthedocs.io/en/stable/install.html#system-requirements] make the same point:
> Certbot is meant to be run directly on a web server, normally by a system administrator. In most cases, running Certbot on your personal computer is not a useful option. The instructions below relate to installing and running Certbot on a server.

On the host is probably best. We could still access the env vars without duplication by sourcing the `.env` if needed.
